src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -336,12 +345,22 @@ private bool IsRecovering(ServerErrorCode code, string message)
 
             return false;
         }
-        
+
         private bool ShouldClearConnectionPool(Exception ex)
         {
             return ex is MongoAuthenticationException;
         }
 
+        private bool ShouldClearConnectionPoolForChannelException(Exception ex, SemanticVersion serverVersion)
```

Comment:
We already have `ShouldClearConnectionPool` method (line 349), not sure maybe we can merge them? Should we validate `NotMaster` error by this ticket rules when we call `GetChannel`?